src/transformers/models/beit/configuration_beit.py

```diff
@@ -93,10 +93,8 @@ class BeitConfig(PretrainedConfig):
             Whether to concatenate the output of the auxiliary head with the input before the classification layer.
         semantic_loss_ignore_index (`int`, *optional*, defaults to 255):
             The index that is ignored by the loss function of the semantic segmentation model.
-        legacy_output (`bool`, *optional*, defaults to `False`):
-            Whether to return the legacy outputs or not (with logits of shape `height / 4 , width / 4`)
-
-            This argument is only present for backward compatibility reasons and will be removed in v5 of Transformers.
+        resize_logits (`bool`, *optional*, defaults to `False`):
+            Whether to resize the logits to the same size as the `pixel_values` or not.
```

Comment:
I'm not sure if this is useful.

Why would people want to resize the logits to the size of the `pixel_values` (which is typically something like 512x512 or 640x640 i.e. square)? All they care about is resize the logits to the size of the original image.

Reply:
Yes but if they are using a custom loss, they can interpolate themselves if they want?

Reply:
They can also write the whole model themselves if they want. The goal of the library is to provide them with things to make it easier ;-)

Reply:
Yes but I wouldn't add this in the same spirit as why we don't add custom losses. If people want to use this model with a custom loss, they can easily use our model implementation to get the `logits`, then do whatever they want with it.

They don't need to re-implement the whole model for that.

Reply:
I agree with @NielsRogge , the only use case I can come up with is a custom loss, and usually, the interpolation is done in the loss directly. Moreover, if we had the ability to resize the logits we should also be sure the user can decide *how* the logits are interpolated ("bilinear" vs something else)

Reply:
To expand my point, suppose people want to get the `logits` the same size as the `pixel_values`. Here's how they can achieve that without having to override the entire model:
```
import transformers import SegformerPreTrainedModel, SegformerForSemanticSegmentation
import torch.nn as nn

class CustomSegFormer(SegformerPreTrainedModel):
    def __init__(self, config):
        super().__init__(config)
        self.segformer = SegformerForSemanticSegmentation.from_pretrained("...")

    def forward(self, pixel_values, labels=None):
        # forward to get raw logits
        outputs = self.segofmrer(ids, attention_mask=mask, token_type_ids=token_type_ids)
        logits = outputs.logits
        
        # next, do whatever they want with the logits to compute loss
        loss = None
        if labels is not None:
               # upsample logits here
               # calculate custom loss here

        return output
```

Reply:
Not a vision engineer so probably not seeing every use case, but I'd favor minimal changes to the output of the model and to let users do what they wish with the output of the model. Resizing logits so that they can be fed to a custom loss of theirs doesn't look like it adds too much complexity and they'll already be computing the loss outside of the model.

Reply:
Removed that bit!